src/tests/multihost/ad/test_idmap.py

```diff
@@ -0,0 +1,65 @@
+""" AD idmap tests from bash
+:requirement: IDM-SSSD-REQ: idmap
+:casecomponent: sssd
+:subsystemteam: sst_idm_sssd
+:upstream: yes
+:status: approved
+"""
+import pytest
+
+from sssd.testlib.common.utils import sssdTools
+
+
+@pytest.mark.usefixtures('joinad', 'create_idmap_users_groups')
+@pytest.mark.tier1_3
+@pytest.mark.idmap
+class Testidmap(object):
+    """ Test cases for idmap
+    :setup:
+        1. Join to AD using realm command.
+        2. Create a non-POSIX user and a non-POSIX group
+        3. Create a user and a group with posix attributes
+    """
+    @staticmethod
+    def test_001_idmap_disable(multihost):
```

Comment:
Either use backupsssdconf fixture or backup the config before changing it and restore it before asserts.
We do not want to have pieces of config for one test to leak to other tests.
Other way may be to use adjoin fixture that have a test scope so the ad is joined with fresh config for every test.

Reply:
backupsssdconf fixture failed when tried first. as by default sssd.conf is not present. the disjoining client from domain removes the sssd.conf as well.  In stead of fixture, I've added client.backup_sssd_conf()

Reply:
ack